pkg/scheduler/apis/config/v1alpha1/conversion.go

```diff
@@ -53,8 +53,11 @@ func Convert_v1alpha1_KubeSchedulerConfiguration_To_config_KubeSchedulerConfigur
 		}
 	}
 	if in.HardPodAffinitySymmetricWeight != nil {
-		args := interpodaffinity.Args{HardPodAffinityWeight: in.HardPodAffinitySymmetricWeight}
-		plCfg := plugins.NewPluginConfig(interpodaffinity.Name, args)
+		encoding := fmt.Sprintf(`{"hardPodAffinityWeight":%d}`, *in.HardPodAffinitySymmetricWeight)
```

Comment:
don't create JSON with string encoding... encode the actual typed struct

Reply:
Actually, we won't need this. It's going away now #89298 (approved)

Reply:
It uses the string encoding so that it doesn't introduce a dependency onf `v1alpha2` in `v1alpha1` conversion.

Reply:
why is v1alpha1 converting to v1alpha2?